app/models/spree/order_decorator.rb

```diff
@@ -1,20 +1,24 @@
 Spree::Order.class_eval do
+  # Update the invoice number before transitioning to complete
+  #
+  state_machine.before_transition :to => :complete, :do => :update_invoice_number!
```

Comment:
I would prefer the usage of Ruby 1.9 Hash syntax ;)

Why do you use a `before_transition`? Wouldn't be an `after_transition` more correct? Because we definitely only want an invoice number if the transition is completed, aren't we?